index.src.html

```diff
@@ -3998,19 +4027,29 @@
             [=combined buffer layout/allocationSize=].
     9. Return |combinedLayout|.
 
+
+: <dfn for=VideoFrame>Copy VideoFrame metadata</dfn> (with |metadata|)
+:: 1. Let |metadataCopySerialized| be [$StructuredSerialize$](|metadata|).
+    2. Let |metadataCopy| be [$StructuredDeserialize$](|metadataCopySerialized|, [=the current Realm=]).
+    3. Let |userMetadataCopy| be |metadataCopy|'s {{VideoFrameMetadata/user}}.
+    4. Call [=SetIntegrityLevel=] with |userMetadataCopy| and frozen.
```

Comment:
As far as I can tell, this is a shallow freeze. Is there maybe way to get a deeply frozen deserialization?

Reply:
This is the best I found so far in terms of link to express the idea.
Agreed though that we should have a deep freeze if we can.
I plan to investigate it but if you already know this/have pointers, let me know.